### PR TITLE
[DEVEX-104] Setup Yarn with Plug'n'Play

### PR DESCRIPTION
**Pull request by lucacavallaro**
# Use yarn with Plug'n'Play

Date: 2024-04-15

## Status

Accepted

## Context

**Plug'n'Play** is an installation strategy, alternative to the classic resolution algorithm used by `npm`/`node.js` with `node_modules`, that simplify working on monorepo and large javascript codebases.

For our use cases, `pnp` has two major advantages over the classic approach:

1. An efficient dependency resolution algorithm and a global cache that allows dependency reuse (the developer installs dependencies only once on his machine, regardless of how many projects it is used in).

2. Catch _ghost dependencies_ (dependencies that are not listed in `package.json`), ensuring that each workspace declares all dependencies it uses.

See the documentation for more details https://yarnpkg.com/features/pnp

## Decision

We decided to enable `plug'n'play` with global cache on our yarn setup.

### CODE REVIEW

**Review by gunzip — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -2,29 +2,23 @@
   "name": "monorepo",
   "private": true,
   "scripts": {
-    "typecheck": "turbo typecheck",
-    "build": "turbo build",
-    "test": "turbo test",
-    "lint": "turbo lint",
-    "format": "prettier --write \"**/*.{ts,tsx,md}\"",
```
Comment (balanza) [inline package.json:9]: **suggestion:** ‏I think separated commands might be useful too

**Review by balanza — COMMENTED** [review]
Great @lucacavallaro! Lots of innovation in this PR!

I'm mostly concerned about the `.yarn` folder. It contains a lot of hidden knowledge, from which is hard to find rationale and documentation online (it's a mix of different sources casted into our needs). I'm afraid it's gonna be the _"only the author can touch it"_ part of our codebase.

My advice is trying to add at least the bare documentation with the rationale behinds those files and with the links to guide readers into the right reference.

Besides that, everything is fine for me. If the team doesn't think of it to be a blocker, we can merge.

**Comment by lucacavallaro** [comment]
> Great @lucacavallaro! Lots of innovation in this PR!
> 
> I'm mostly concerned about the `.yarn` folder. It contains a lot of hidden knowledge, from which is hard to find rationale and documentation online (it's a mix of different sources cast into our needs). I'm afraid it's gonna be the _"only the author can touch it"_ part of our codebase.
> 
> My advice is to try to add at least the bare documentation with the rationale behind those files and with the links to guide readers to the right reference.
> 
> Besides that, everything is fine for me. If the team doesn't think of it to be a blocker, we can merge.

As I written in the `README.md` that file are automatically generated by the command `yarn dlx @yarnpkg/sdks vscode`. What you need to now about "sdks" is documented [here](https://yarnpkg.com/getting-started/editor-sdks). It is not important to understand what the source code does, the only thing to keep in mind is that they are files necessary to use yarn + pnp with vscode.

File: package.json (hunk 1 above)
Comment (lucacavallaro) [reply package.json:9]: ok, I'll put them back!

**Comment by balanza** [comment]
> > Great @lucacavallaro! Lots of innovation in this PR!
> > I'm mostly concerned about the `.yarn` folder. It contains a lot of hidden knowledge, from which is hard to find rationale and documentation online (it's a mix of different sources cast into our needs). I'm afraid it's gonna be the _"only the author can touch it"_ part of our codebase.
> > My advice is to try to add at least the bare documentation with the rationale behind those files and with the links to guide readers to the right reference.
> > Besides that, everything is fine for me. If the team doesn't think of it to be a blocker, we can merge.
> 
> As I written in the `README.md` that file are automatically generated by the command `yarn dlx @yarnpkg/sdks vscode`. What you need to now about "sdks" is documented [here](https://yarnpkg.com/getting-started/editor-sdks). It is not important to understand what the source code does, the only thing to keep in mind is that they are files necessary to use yarn + pnp with vscode.

Great! I really missed that sentence I guess! LGTM

**Pull request merged**